pkg/kubectl/cmd/run.go

```diff
@@ -478,8 +478,9 @@ func handleAttachPod(f *cmdutil.Factory, c *client.Client, ns, name string, opts
 	opts.Client = c
 	opts.PodName = name
 	opts.Namespace = ns
+	stderr := opts.Err
 	if err := opts.Run(); err != nil {
-		fmt.Fprintf(opts.Err, "Error attaching, falling back to logs: %v\n", err)
+		fmt.Fprintf(stderr, "Error attaching, falling back to logs: %v\n", err)
```

Comment:
Yeah I probably should undo that bit and find a better way to do this. I
check for nil to determine if we need to handle stderr.

On Wednesday, September 7, 2016, Janet Kuo <notifications@github.com> wrote:

> In pkg/kubectl/cmd/run.go
> <https://github.com/kubernetes/kubernetes/pull/32212#discussion_r77909155>
> :
>
> >  	if err := opts.Run(); err != nil {
> > -		fmt.Fprintf(opts.Err, "Error attaching, falling back to logs: %v\n", err)
> > +		fmt.Fprintf(stderr, "Error attaching, falling back to logs: %v\n", err)
>
> Shall we add a check here to make sure stderr is not nil or we're good?
>
> cc @ncdc <https://github.com/ncdc> why set the opts.Err to nil in
> opts.Run()? It's surprising to the callers IMO.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/32212/files/61605bd930504268f309632b832a34dd96ff7650#r77909155>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAABYp4_nQlZvmebMp4V-yAWUVoksq0bks5qny8ygaJpZM4J3KmP>
> .
>